packages/react/src/BlockNoteView.tsx

```diff
@@ -1,15 +1,52 @@
 import { BlockNoteEditor, BlockSchema } from "@blocknote/core";
+import { MantineProvider } from "@mantine/core";
 import { EditorContent } from "@tiptap/react";
-// import { BlockNoteTheme } from "./BlockNoteTheme";
-// import { MantineProvider } from "@mantine/core";
+import { getBlockNoteTheme } from "./BlockNoteTheme";
+import { ReactNode, useEffect, useState } from "react";
+import { FormattingToolbarPositioner } from "./FormattingToolbar/components/FormattingToolbarPositioner";
+import { HyperlinkToolbarPositioner } from "./HyperlinkToolbar/components/HyperlinkToolbarPositioner";
+import { SlashMenuPositioner } from "./SlashMenu/components/SlashMenuPositioner";
+import { SideMenuPositioner } from "./SideMenu/components/SideMenuPositioner";
 
 export function BlockNoteView<BSchema extends BlockSchema>(props: {
-  editor: BlockNoteEditor<BSchema> | null;
+  editor: BlockNoteEditor<BSchema>;
+  children?: ReactNode;
 }) {
+  const [ready, setReady] = useState(false);
+
+  useEffect(() => {
+    function checkReady() {
+      if (!props.editor.ready) {
+        window.setTimeout(checkReady, 100);
+      } else {
+        setReady(true);
+      }
+    }
+    checkReady();
+
+    // TODO: Shouldn't this work with props.editor.ready in deps? It fails in
+    //  tests so it doesn't seem to.
+    // if (props.editor.ready) {
+    //   setReady(true);
+    // }
+  }, [props.editor.ready]);
+
   return (
-    // TODO: Should we wrap editor in MantineProvider? Otherwise we have to duplicate color hex values.
-    // <MantineProvider theme={BlockNoteTheme}>
-    <EditorContent editor={props.editor?._tiptapEditor || null} />
-    // </MantineProvider>
+    <MantineProvider theme={getBlockNoteTheme()}>
+      <EditorContent
+        editor={props.editor?._tiptapEditor}
+        // TODO: Better class name
+        className={"BlockNote"}>
```

Comment:
Maybe we don't need it since you can add it as a prop to `BlockNoteView`. I figured it might make sense to add some kind of default class otherwise it's not immediately obvious that it's a wrapper for BlockNote.